pkg/container/docker.go

```diff
@@ -184,7 +199,20 @@ func (f *BuilderFactory) GenerateImageHash() (string, error) {
 		return "", ErrHashingBuildContext.Wrap(err)
 	}
 
-	logrus.Debug("Generated image hash: ", fmt.Sprintf("%x", hasher.Sum(nil)))
+	f.logger.Debug("Generated image hash: ", fmt.Sprintf("%x", hasher.Sum(nil)))
 
 	return fmt.Sprintf("%x", hasher.Sum(nil)), nil
 }
+
+func verifyOptions(opts BuilderFactoryOptions) error {
```

Comment:
Shouldn't image builder be validated here? Since we check for it explicitly in multiple places?

Adding the check here should make it impossible to create the BuilderFactory without it, so we could clean up all those inline checks.

Reply:
that could be neater, however the image builder is not a requirement as the user can set an image that is already built. Image builder is required only if we want to build from git or from a local directory.
Do you think we should make it a requirement?

Reply:
To preface, If this is a bigger change we resolve into a follow up issue to unblock this PR. 

It sounds like there are 2 distinct user flows. 
1. No image exists yet, we need to build it
2. An image exists, and we need to use it.

So it would seem that the more concrete thing that needs to be check is the built image itself. In absence of that, it seems we are checking the name or the image builder as proxies. 

Is there a way to know if the image was already built or it needs to be built? That might clean up the logic here to make it more specific to each scenario vs having some of the context hidden.

Either way, like i said to start, if this is a deeper rabbit hole we can discuss this more in an issue or offline and unblock this PR.

Reply:
I think it is covered in the #525 as it requires a refactor around the image names and the main focus of this PR is to introduce logger to the builderFactory. Therefore, I agree with the option that we unlock this PR and track it in the other issue.